### ISSUE TITLE
GCC 5.4 complains -Werror=int-to-pointer-cast when

### ISSUE THREAD

**Issue by liuqun** [intel/tpm2-abrmd#239]
GCC 5.4 complains -Werror=int-to-pointer-cast when mocka test unit is enabled by "./configure --enable-unit"

https://github.com/intel/tpm2-abrmd/blob/bc1b31e7dab08f5d6151b0d72816fcce3d6eb022/test/command-attrs_unit.c#L114

https://github.com/intel/tpm2-abrmd/blob/bc1b31e7dab08f5d6151b0d72816fcce3d6eb022/test/command-attrs_unit.c#L145

```
$ ./bootstrap && ./configure --enable-unit && make check

...

test/command-attrs_unit.c: In function ‘__wrap_access_broker_lock_sapi’:
test/command-attrs_unit.c:114:12: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
     return (TSS2_SYS_CONTEXT*)mock ();
            ^
test/command-attrs_unit.c: In function ‘__wrap_Tss2_Sys_GetCapability’:
test/command-attrs_unit.c:145:21: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
     command_attrs = (TPMA_CC*)mock ();
                     ^
cc1: all warnings being treated as errors
```

```
gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/arm-linux-gnueabihf/5/lto-wrapper
Target: arm-linux-gnueabihf
Configured with: ../src/configure -v --with-pkgversion='Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.5' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libitm --disable-libquadmath --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-armhf/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-armhf --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-armhf --with-arch-directory=arm --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --enable-multilib --disable-sjlj-exceptions --with-arch=armv7-a --with-fpu=vfpv3-d16 --with-float=hard --with-mode=thumb --disable-werror --enable-multilib --enable-checking=release --build=arm-linux-gnueabihf --host=arm-linux-gnueabihf --target=arm-linux-gnueabihf
Thread model: posix
gcc version 5.4.0 20160609 (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.5) 
```

Tested on Ubuntu Core 16 for Raspberry Pi 3 (ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-)
```
$ uname -a
Linux localhost.localdomain 4.4.0-1030-raspi2 #37-Ubuntu SMP Thu Oct 20 15:06:25 UTC 2016 armv7l armv7l armv7l GNU/Linux

$ gcc --version
gcc (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.5) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by flihp** [comment]
should be resolved in #241

**Comment by flihp** [comment]
I've merged #241 into the 1.x branch. If this didn't resolve the issue please re-open this ticket or create a new one.
